packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -3109,13 +3109,17 @@ void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t, mj_node_t>::
   });
 
   // round the target part weights.
-  Kokkos::parallel_for(
-    Kokkos::RangePolicy<typename mj_node_t::execution_space, mj_part_t>
-      (0, num_cuts + 1),
-    KOKKOS_LAMBDA (mj_part_t i) {
-    current_target_part_weights(i) =
-      long(current_target_part_weights(i) + 0.5);
-  });
+  // Note need to discuss regarding DragonFly commits and determine if we
+  // would not simply check mj_uniform_weights here.
+  if (!bUniformPartsCheck || this->mj_uniform_weights[0]) {
```

Comment:
This looks good to me for nonuniform partitioning. It may have been rounding unnecessarily previously.